### ISSUE TITLE
Pore and throat volume models are not rigorous

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#2558]
We offer `op.models.geometry.pore_volume.sphere` and `op.models.geometry.throat_volume.cylinder`, but if these are used together they do not properly account for the overlap.  We have models for computing this (`op.models.geometry.throat_volume.lens`), but these are not included in the `spheres_and_cylinders` model collection.  More problematically, to properly account for this we'd need to know the throat endpoints, which has actually be removed.  

So basically, the `spheres_and_cylinders` collection should have a 'throat.full_volume' model which is the volume of a cylinder, and a `'throat.volume'` model which is the full volume less then lens volumes.
